### PR TITLE
Light refactor of other records, bringing in line

### PR DESCRIPTION
**Pull request by lcpriest**

### CODE REVIEW

File: lib/gecko.rb [hunk 1]
```diff
@@ -30,3 +30,4 @@
 require 'gecko/record/purchase_order_line_item'
 require 'gecko/record/tax_type'
 require 'gecko/record/payment_term'
+require 'gecko/record/note'
```
Comment (bradleypriest) [inline lib/gecko.rb]: Can we change hound here please

File: lib/gecko/record/note.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+require 'gecko/record/base'
+
+module Gecko
```
Comment (bradleypriest) [inline lib/gecko/record/note.rb]: I'd like to not encourage the whole Notes API until we have a chance to rebuild it from scratch

File: lib/gecko/record/product.rb [hunk 3]
```diff
@@ -12,12 +12,12 @@ class Product < Base
       attribute :opt1,         String
       attribute :opt2,         String
       attribute :opt3,         String
-      attribute :tags,         String
+      attribute :tags,         Array[String]
```
Comment (bradleypriest) [inline lib/gecko/record/product.rb]: Really?

File: lib/gecko/record/product.rb [hunk 4]
```diff
@@ -12,12 +12,12 @@ class Product < Base
...
 
       attribute :status,       String
 
       attribute :image_url,    String,  readonly: true
-      # attribute :quantity,     Integer, readonly: true
+      attribute :quantity,     Integer, readonly: true
```
Comment (bradleypriest) [inline lib/gecko/record/product.rb]: This is private

File: lib/gecko/record/purchase_order.rb [hunk 5]
```diff
@@ -4,7 +4,8 @@ module Gecko
   module Record
     class PurchaseOrder < Base
       has_many :purchase_order_line_items
-      has_many :procurements
+      # has_many :procurements
+      attribute :procurement_ids,     Array
```
Comment (bradleypriest) [inline lib/gecko/record/purchase_order.rb]: Is this necessary?

File: lib/gecko/record/user.rb [hunk 6]
```diff
@@ -16,10 +16,10 @@ class User < Base
       attribute :status,             String,   readonly: true
       attribute :billing_contact,    Boolean,  readonly: true
 
-      # attribute :sales_report_email, Boolean
-      # attribute :action_items_email, String
+      attribute :sales_report_email, Boolean
```
Comment (bradleypriest) [inline lib/gecko/record/user.rb]: I'd rather these were marked as private and handled via UI

**Comment by lcpriest** [comment]
Waiting on #55

File: lib/gecko/record/product.rb (hunk 4 above)
Comment (lcpriest) [inline lib/gecko/record/product.rb]: Private == deleted from gem entirely?

Comment (bradleypriest) [inline lib/gecko/record/product.rb]: :+1:

File: lib/gecko/record/purchase_order.rb (hunk 5 above)
Comment (lcpriest) [inline lib/gecko/record/purchase_order.rb]: Nope, just read the API more

File: lib/gecko/record/product.rb (hunk 3 above)
Comment (lcpriest) [inline lib/gecko/record/product.rb]: We use the same syntax on the order.
https://github.com/tradegecko/gecko/commit/b5e7424eed8804c7af63813eef1e7a4825c8e181

**Comment by tradeweeyoung** [comment]
LGTM

File: lib/gecko/record/product.rb [hunk 7]
```diff
@@ -12,13 +12,11 @@ class Product < Base
       attribute :opt1,         String
       attribute :opt2,         String
       attribute :opt3,         String
-      attribute :tags,         String
+      attribute :tags,         Array[String]
 
       attribute :status,       String
 
       attribute :image_url,    String,  readonly: true
-      # attribute :quantity,     Integer, readonly: true
```
Comment (levady) [inline lib/gecko/record/product.rb]: This attribute and the one below it is deprecated? If so then shouldn't be like this?

```ruby
## DEPRECATED
# attribute :quantity,     Integer, readonly: true
# attribute :search_cache, String,  readonly: true

**Comment by levady** [comment]
Other than 2 hound errors and 1 minor comment, LGTM.

File: lib/gecko/record/product.rb (hunk 7 above)
Comment (lcpriest) [inline lib/gecko/record/product.rb]: The rule I followed was, if it were previously commented, I can remove it, if it was available but we are deprecating it, comment it out.

**Comment by lcpriest** [comment]
@bradleypriest can I merge?

File: lib/gecko/record/product.rb [hunk 8]
```diff
@@ -17,8 +17,6 @@ class Product < Base
...
-      # attribute :search_cache, String,  readonly: true
```
Comment (bradleypriest) [inline lib/gecko/record/product.rb]: Can we remove this from the actual API as well, I don't think anyone is using it, it was a failed test from a long, long time ago

File: lib/gecko/record/purchase_order.rb [hunk 9]
```diff
@@ -29,7 +30,11 @@ class PurchaseOrder < Base
 
       attribute :total,                 BigDecimal, readonly: true
       attribute :cached_quantity,       BigDecimal, readonly: true
-      attribute :cached_total,          BigDecimal, readonly: true
+
+      attribute :tags,                  Array[String]
+
+      # DEPRECATED
+      # attribute :cached_total,          BigDecimal, readonly: true
```
Comment (bradleypriest) [inline lib/gecko/record/purchase_order.rb]: Deprecated?

Comment (lcpriest) [inline lib/gecko/record/purchase_order.rb]: Is it not?

File: lib/gecko/record/product.rb (hunk 8 above)
Comment (lcpriest) [inline lib/gecko/record/product.rb]: Yes

File: lib/gecko/record/purchase_order.rb (hunk 9 above)
Comment (bradleypriest) [inline lib/gecko/record/purchase_order.rb]: I'd love to have total on an order purchase order but we should actually just call it total

Comment (lcpriest) [inline lib/gecko/record/purchase_order.rb]: It is called that, two lines up

Comment (bradleypriest) [inline lib/gecko/record/purchase_order.rb]: Ok, good, can you please clean up Invoice/StockAdjustment to just total in our API as well?

File: lib/gecko/record/fulfillment.rb [hunk 10]
```diff
@@ -21,8 +23,6 @@ class Fulfillment < Base
       attribute :shipped_at,       DateTime
       attribute :received_at,      DateTime
       attribute :service,          String
-
-      attribute :receipt,          Hash
```
Comment (bradleypriest) [inline lib/gecko/record/fulfillment.rb]: Is anyone using this?

Comment (lcpriest) [inline lib/gecko/record/fulfillment.rb]: I don't know. Is it something we want in our API?

File: lib/gecko/record/purchase_order.rb (hunk 9 above)
Comment (lcpriest) [inline lib/gecko/record/purchase_order.rb]: Also order?

File: lib/gecko/record/company.rb [hunk 11]
```diff
@@ -29,8 +29,10 @@ class Company < Base
 
       attribute :tax_number,             String
 
-      attribute :default_tax_rate,       BigDecimal
       attribute :default_discount_rate,  BigDecimal
+
+      # DEPRECATED
+      # attribute :default_tax_rate,       BigDecimal
```
Comment (bradleypriest) [inline lib/gecko/record/company.rb]: Isn't it just readonly, or do we plan to remove it?

Comment (lcpriest) [inline lib/gecko/record/company.rb]: I planned to remove it as tax_type exists and in most cases doesn't actually need to be used as it's handled on our end.

Comment (lcpriest) [inline lib/gecko/record/company.rb]: @bradleypriest my mistake, we don't use this in B2B, we use the effective_tax_rate on the tax_type model.

Comment (lcpriest) [inline lib/gecko/record/company.rb]: This is useful though, it's an alias of default_tax_type#effective_tax_rate

File: lib/gecko/record/purchase_order.rb (hunk 9 above)
Comment (lcpriest) [inline lib/gecko/record/purchase_order.rb]: Done

**Pull request merged**